keras_cv/models/backbones/csp_darknet/csp_darknet_backbone.py

```diff
@@ -0,0 +1,385 @@
+# Copyright 2023 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""DarkNet models for KerasCV. """
+import copy
+
+from tensorflow import keras
+from tensorflow.keras import layers
+
+from keras_cv.models import utils
+from keras_cv.models.backbones.backbone import Backbone
+from keras_cv.models.backbones.csp_darknet.csp_darknet_backbone_presets import (
+    backbone_presets,
+)
+from keras_cv.models.backbones.csp_darknet.csp_darknet_backbone_presets import (
+    backbone_presets_with_weights,
+)
+from keras_cv.models.backbones.csp_darknet.darknet_utils import (
+    CrossStagePartial,
+)
+from keras_cv.models.backbones.csp_darknet.darknet_utils import DarknetConvBlock
+from keras_cv.models.backbones.csp_darknet.darknet_utils import (
+    DarknetConvBlockDepthwise,
+)
+from keras_cv.models.backbones.csp_darknet.darknet_utils import Focus
+from keras_cv.models.backbones.csp_darknet.darknet_utils import (
+    SpatialPyramidPoolingBottleneck,
+)
+from keras_cv.utils.python_utils import classproperty
+
+
+@keras.utils.register_keras_serializable(package="keras_cv.models")
+class CSPDarkNetBackbone(Backbone):
+    """This class represents the CSPDarkNet architecture.
+
+    Reference:
+        - [YoloV4 Paper](https://arxiv.org/abs/1804.02767)
+        - [CSPNet Paper](https://arxiv.org/pdf/1911.11929)
+        - [YoloX Paper](https://arxiv.org/abs/2107.08430)
+        - [YoloX implementation](https://github.com/ultralytics/yolov3)
+
+    For transfer learning use cases, make sure to read the
+    [guide to transfer learning & fine-tuning](https://keras.io/guides/transfer_learning/).
+
+    Args:
+        channels: A list of ints, the number of channels for each dark level in
+            the model.
+        depths: A list of ints, the depth for each dark level in the model.
+        include_rescaling: bool, whether to rescale the inputs. If set to
```

Comment:
nit: this is called a `bool` while the one below is called a `boolean`. I don't care which one we use as long as we're consistent.